tqueue.cc

```diff
@@ -82,15 +70,14 @@ bool    operator <(const Queue_entry &q1, const Queue_entry &q2) {
 int Time_queue::remove(
     Time_sensitive *obj
 ) {
-	if (data.empty())
+	auto toRemove = std::find_if(data.begin(), data.end(),
+			[obj](const auto& el) { return el.handler == obj; });
+	if (toRemove == data.end()) {
 		return 0;
-	for (auto it = data.begin();
-	        it != data.end(); ++it) {
-		if (it->handler == obj) {
-			obj->queue_cnt--;
-			data.erase(it);
-			return 1;
-		}
+	} else {
+		(*toRemove).handler->queue_cnt--;
+		data.erase(toRemove);
+		return 1;
 	}
```

Comment:
Please avoid using `else` here, as the `if` is just a return. Maybe invert the condition check? See also dead code comment.